gui

```diff
@@ -191,570 +438,258 @@ runonce <<"EOF"
+    elif [ -f /tmp/terminalexit ];then #if terminal doesn't start in 3 seconds, then /tmp/terminalexit will exist.
+      #revert back to yad
+      echo 'yad-default' > "{DIRECTORY}/data/settings/App List Style"
+      error "The xlunch compilation terminal exited prematurely."
     else
-      echo -e "\nClose this window to exit."
-      read enter #technically you could press Enter to exit.
+      sleep 1
     fi
-    ' "$title"
-  
-  #re-preload all categories in background
-  "${DIRECTORY}/etc/preload-daemon" "$guimode" &>/dev/null &
-}
-
-#Determine message of the day. If announcements file missing or over a day old, download it.
-if [ ! -f "${DIRECTORY}/data/announcements" ] || [ ! -z "$(find "${DIRECTORY}/data/announcements" -mtime +1 -print)" ]; then
-  wget https://raw.githubusercontent.com/Botspot/pi-apps-announcements/main/message -qO "${DIRECTORY}/data/announcements"
-fi
-
-#app list mode. Allowed values: 'yad', 'xlunch'
-guimode="$(cat "${DIRECTORY}/data/settings/App List Style")"
-[ -z "$guimode" ] && guimode=yad-default
-#mode=yad
-
-#Variable 1 is used to view the details of the specified app
-if [ ! -z "$1" ];then
-  app="$1"
-  action='details'
-else
-  action='main-window'
+  done
 fi
-prefix=''
 
-while true;do
+if [[ "$guimode" == yad* ]];then
   
-  [ ! -z "$prefix" ] && echo "Prefix: $prefix"
-  echo "App: ${app}"
-  echo "Action: $action"
-  echo "Prevaction: $prevaction"
-  
-  #most actions need the app variable to be filled. These don't: main-window, search, exit, back, unknown, view-updates
-  if [ -z "$app" ] && ! echo "$action" | grep -q 'main-window\|search\|exit\|back\|unknown\|view-updates';then
-    action='mind-reading'
-    echo "Action has been changed to 'mind-reading'"
-  fi
+  #create a named pipe to send app list through to yad
+  pipe="$(mktemp -u)" #get a random filename to work with
+  mkfifo $pipe #make the named pipe
+  trap "rm $pipe" EXIT #remove this named pipe on exit
+  echo pipe is $pipe
   
-  if [ "$action" == main-window ];then
+  while true;do #main operating loop for yad window
+    #in background process, send initial app list to yad; first echo line waits until yad is accepting input, then use small delay to prevent items from being missed
+    (echo -e '\f' > $pipe
+    sleep 0.5
+    "${DIRECTORY}/preload" yad '' >> $pipe) &
     
-    prevaction="$action" #remember the previous action we came from
+    #constantly read from the pipe and send all output to yad
+    #if an item is selected, this script will be run with the row's contents
+    tail -F $pipe | yad --center --title='Pi-Apps' \
+      --image="${DIRECTORY}/icons/logo-64.png" --image-on-top \
+      --list --no-headers --column=:IMG --column=:IMG --column=Name --column=Sysname:HD --column=tip:HD \
+      --separator='\n' --window-icon="${DIRECTORY}/icons/logo.png" \
+      --print-column=4 --tooltip-column=5 \
+      --dclick-action=true --select-action "env pipe=$pipe geometry2=$geometry2 '$0'" \
```

Comment:
```suggestion
      --dclick-action "env pipe=$pipe geometry2=$geometry2 '${DIRECTORY}/gui'" --select-action "env pipe=$pipe geometry2=$geometry2 ''${DIRECTORY}/gui''" \
```

this makes it so double clicking an app still opens the window. for example, if the user clicks on an app, the window opens, and if they they unfocus that window it will close. if they want to open it again, they have to click on a DIFFERENT app then back to the app they are interested in. This fixes that so now they can just double click on the app and it will open again (should be intuitive)